api/notifications/notifications.py

```diff
@@ -54,18 +55,23 @@ def create_action_notifications(action):
     elif action.action_status == ActionStatus.ASSIGNED:
 
         # 1. Let the assigned volunteer know.
-        # If this notification hasn't been sent before, or if it was previously
-        # sent to someone else, we should send it to the assigned volunteer.
-        if not notification_exists(action, NotificationTypes.VOLUNTEER_ASSIGNED) \
-            or action.assigned_volunteer.email != get_latest_notification(action,
-                                                                          NotificationTypes.VOLUNTEER_ASSIGNED).recipients[0]:
+
+        # Use `in` as the value might be `None`, in case of first assignment
+        if 'assigned_volunteer_id' in changed:
             create([action.assigned_volunteer.email], action=action,
                    notification_type=NotificationTypes.VOLUNTEER_ASSIGNED)
 
-        # 2. Let those not assigned know.
-        # Only send a notification to volunteers that have not received this email for this action.
+        # 2. Let the previously assigned volunteer know
+            if (changed.get('assigned_volunteer_id')):
```

Comment:
We've already checked `if 'assigned_volunteer_id' in changed`, so we shouldn't need to check again.

Reply:
Removed repeated check

Reply:
Undone - this _was_ needed.

Reply:
My bad, I should have put a comment. The first one checked if there was a change of assigned volunteer, the second if there was actually a previous volunteer to notify.